### ISSUE TITLE
java.lang.IllegalArgumentException: x not finite

### ISSUE THREAD

**Issue by andris155** [SirBlobman/CombatLogX#80]
**Describe the bug**
java.lang.IllegalArgumentException: x not finite

**To Reproduce**
How can I see the issue for myself? Please explain how it happened.
I don't know. Find this error console.

**Expected behavior**
No write this error?

**Screenshots**

**Minecraft Version**
 - Do `/clx version` to find out
20.07 14:57:53 [Server] INFO Full Version: git-Paper-639 (MC: 1.13.2)
20.07 14:57:53 [Server] INFO Bukkit Version: 1.13.2-R0.1-SNAPSHOT
20.07 14:57:53 [Server] INFO Minecraft Version: 1.13.2
20.07 14:57:53 [Server] INFO NMS Version: 1_13_R2
20.07 14:57:53 [Server] INFO CombatLogX by SirBlobman
20.07 14:57:53 [Server] INFO Latest Version: v9.8.4.0
20.07 14:57:53 [Server] INFO Installed Version: v9.8.3.0
20.07 14:57:53 [Server] INFO Getting expansion versions...
20.07 14:57:53 [Server] INFO Towny Compatibility v14.2
20.07 14:57:53 [Server] INFO Cheat Prevention v14.8
20.07 14:57:53 [Server] INFO WorldGuard Compatibility v14.8

**CombatLogX Version:**
 - Do `/clx version` to find out
20.07 14:57:53 [Server] INFO Full Version: git-Paper-639 (MC: 1.13.2)
20.07 14:57:53 [Server] INFO Bukkit Version: 1.13.2-R0.1-SNAPSHOT
20.07 14:57:53 [Server] INFO Minecraft Version: 1.13.2
20.07 14:57:53 [Server] INFO NMS Version: 1_13_R2
20.07 14:57:53 [Server] INFO CombatLogX by SirBlobman
20.07 14:57:53 [Server] INFO Latest Version: v9.8.4.0
20.07 14:57:53 [Server] INFO Installed Version: v9.8.3.0
20.07 14:57:53 [Server] INFO Getting expansion versions...
20.07 14:57:53 [Server] INFO Towny Compatibility v14.2
20.07 14:57:53 [Server] INFO Cheat Prevention v14.8
20.07 14:57:53 [Server] INFO WorldGuard Compatibility v14.8

**Additional context**
```
[11:44:27] [Server thread/WARN]: [CombatLogX] Task #4649538 for CombatLogX v9.8.3.0 generated an exception
java.lang.IllegalArgumentException: x not finite
	at org.bukkit.util.NumberConversions.checkFinite(NumberConversions.java:118) ~[patched_1.13.2.jar:git-Paper-639]
	at org.bukkit.util.Vector.checkFinite(Vector.java:800) ~[patched_1.13.2.jar:git-Paper-639]
	at org.bukkit.craftbukkit.v1_13_R2.entity.CraftEntity.setVelocity(CraftEntity.java:268) ~[patched_1.13.2.jar:git-Paper-639]
	at com.SirBlobman.expansion.towny.listener.ListenTowny.lambda$preventEntry$0(ListenTowny.java:95) ~[?:?]
	at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftTask.run(CraftTask.java:84) ~[patched_1.13.2.jar:git-Paper-639]
	at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:449) ~[patched_1.13.2.jar:git-Paper-639]
	at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:1010) ~[patched_1.13.2.jar:git-Paper-639]
	at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:439) ~[patched_1.13.2.jar:git-Paper-639]
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:940) ~[patched_1.13.2.jar:git-Paper-639]
	at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:837) ~[patched_1.13.2.jar:git-Paper-639]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by SirBlobman** [comment]
I need more information please.

When does this error happen? How can I see it for myself?

**Comment by ArtelGG** [comment]
Just received a similar error.
```
26.07 17:38:11 [Server] WARN [CombatLogX] Task #157831 for CombatLogX v9.8.4.0 generated an exception
26.07 17:38:11 [Server] INFO java.lang.IllegalArgumentException: x not finite
26.07 17:38:11 [Server] INFO at org.bukkit.util.NumberConversions.checkFinite(NumberConversions.java:118) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at org.bukkit.util.Vector.checkFinite(Vector.java:800) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at org.bukkit.craftbukkit.v1_14_R1.entity.CraftEntity.setVelocity(CraftEntity.java:407) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at com.SirBlobman.expansion.worldguard.listener.ListenWorldGuard.lambda$preventEntry$0(ListenWorldGuard.java:116) ~[?:?]
26.07 17:38:11 [Server] INFO at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftTask.run(CraftTask.java:84) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:452) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.b(MinecraftServer.java:1147) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at net.minecraft.server.v1_14_R1.DedicatedServer.b(DedicatedServer.java:420) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:1074) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:918) ~[patched_1.14.4.jar:git-Paper-146]
26.07 17:38:11 [Server] INFO at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

**Comment by SirBlobman** [comment]
> Just received a similar error.
> 
> ```
> 26.07 17:38:11 [Server] WARN [CombatLogX] Task #157831 for CombatLogX v9.8.4.0 generated an exception
> 26.07 17:38:11 [Server] INFO java.lang.IllegalArgumentException: x not finite
> 26.07 17:38:11 [Server] INFO at org.bukkit.util.NumberConversions.checkFinite(NumberConversions.java:118) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at org.bukkit.util.Vector.checkFinite(Vector.java:800) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at org.bukkit.craftbukkit.v1_14_R1.entity.CraftEntity.setVelocity(CraftEntity.java:407) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at com.SirBlobman.expansion.worldguard.listener.ListenWorldGuard.lambda$preventEntry$0(ListenWorldGuard.java:116) ~[?:?]
> 26.07 17:38:11 [Server] INFO at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftTask.run(CraftTask.java:84) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:452) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.b(MinecraftServer.java:1147) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at net.minecraft.server.v1_14_R1.DedicatedServer.b(DedicatedServer.java:420) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:1074) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:918) ~[patched_1.14.4.jar:git-Paper-146]
> 26.07 17:38:11 [Server] INFO at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
> ```

If you're having the same issue I need your versions and an explanation of when it happens please.

**Comment by ArtelGG** [comment]
Server version is PaperSpigot 1.14.4-b146, you can see the plugin version in the stack trace.
I have no idea what caused it to be set off. Appeared to just be random from what I could tell.

**Comment by SirBlobman** [comment]
Should be fixed in 9.8.4.1, re-open this if its not
